File: python/web.py
import os
import redis
from urlparse import urlparse
from bottle import route, run, response, abort

redisUrl = urlparse(os.environ['REDISTOGO_URL'])
pool = redis.ConnectionPool(host=redisUrl.hostname, port=redisUrl.port, db=0, password=redisUrl.password)

@route('/quote/<author>')
def get_quote(author):
    redisClient = redis.Redis(connection_pool=pool)
    redisKey = 'quote:' + author
    result = redisClient.srandmember(redisKey)
    response.content_type = 'application/json'
    if result:
        return result
    else:
        abort(404)

run(host='localhost', port=8080, debug=False)

File: ruby/web.rb
require 'sinatra'
require 'redis'
require 'uri'

set :logging, false
set :port, 80
set :bind, '0.0.0.0'
set :environment, :production

redisUrl = URI(ENV['REDISTOGO_URL'])
redis = Redis.new(:driver => :hiredis,
                  :host => redisUrl.host,
                  :port => redisUrl.port,
                  :password => redisUrl.password)

get '/quote/:author' do |author|
    redisKey = 'quote:' + author
    content_type 'application/json'
    result = redis.srandmember(redisKey)
    halt 404 if result == nil
    result
end
